src/seedu/addressbook/data/person/Block.java

```diff
@@ -8,8 +10,11 @@
 public class Block {
     private String block;
 
-    public Block(String block) {
-        this.block = block;
+    public static String REGEX = "(^[\\d]{1,4}[A-Z]{0,1}$)|(^$)";
```

Comment:
i m not sure what this regex is doing! i got confused trying to work through :-P
may be this warrants a comment in the code as to what it does (you are the better judge here)